### PR TITLE
add support for company-kind

### PR DESCRIPTION
**Pull request by jidibinlin**
If we add company-kind support. completion front end such as corfu can get the kind information to implement more feature. 
For my purposes I can use it to make corfu support icon display when pop up the completion frame. 

image

### CODE REVIEW

**Comment by joaotavora** [comment]
Looks good in principle, but please remove unrelated whitespace changes first.

**Comment by jidibinlin** [comment]
Alright, But a suggest you automatically delete whitespace when you save buffer. I will remove unrelated whitespace changes.

**Comment by jidibinlin** [comment]
@joaotavora  please check it. Happy hacking.

**Comment by joaotavora** [comment]
No it needs better commit messages and better formatting. I don't have time right now so be patient. The idea is good. Maybe you can guess from other commits the coding standards

File: lib/sly-completion.el [hunk 1]
```diff
@@ -291,6 +291,11 @@ ANNOTATION) describing each completion possibility."
                              (when suggestion
                                (delete-region (- (point) (length obj)) (point))
                                (insert suggestion))))
+          :company-kind (lambda (str)
+                          (let* ((annotation (sly-completion-annotation str))
+                                 (kind (car (split-string annotation " " t))))
+                            (intern kind)
+                            ))
```
Comment (toniz4) [inline lib/sly-completion.el:298]: The parenthesis should close in the same line

**Comment by joaotavora** [comment]
Fairly sure, this was fixed in https://github.com/joaotavora/sly/pull/634.  So closing.  Thanks.

**Pull request closed without merging**